src/util/charmrun-src/charmrun.C

```diff
@@ -5482,16 +5484,16 @@ struct local_nodestart
       switch (proc_per.unit())
       {
         case Unit::Host:
-          sprintf(envp[envc + n], "CmiProcessPerHost=%d", proc_per.host);
+          snprintf(envp[envc + n], 256, "CmiProcessPerHost=%d", proc_per.host);
           break;
         case Unit::Socket:
-          sprintf(envp[envc + n], "CmiProcessPerSocket=%d", proc_per.socket);
+          snprintf(envp[envc + n], 256, "CmiProcessPerSocket=%d", proc_per.socket);
           break;
         case Unit::Core:
-          sprintf(envp[envc + n], "CmiProcessPerCore=%d", proc_per.core);
+          snprintf(envp[envc + n], 256, "CmiProcessPerCore=%d", proc_per.core);
           break;
         case Unit::PU:
-          sprintf(envp[envc + n], "CmiProcessPerPU=%d", proc_per.pu);
+          snprintf(envp[envc + n], 256, "CmiProcessPerPU=%d", proc_per.pu);
           break;
         default:
```

Comment:
A lot of `256`s here (I count at least 15) could use `static constexpr int envlen = 256;`